core/global.d.ts

```diff
@@ -116,7 +116,7 @@ declare function error(message: string, level?: number): never;
  * metatable has a __metatable field, returns the associated value. Otherwise,
  * returns the metatable of the given object.
  */
-declare function getmetatable<T extends object>(object: T): LuaMetatable<T> | undefined;
+declare function getmetatable<T extends object>(object: T): LuaMetatable<T, object> | undefined;
```

Comment:
It might be better to make second type parameter on `LuaMetatable` optional (with default of `object`). It should make more sense when `LuaMetatable` is used directly and would avoid a breaking change.